### PR TITLE
feat: css realtime preview

### PR DESCRIPTION
**Pull request by Innei**
### <samp>🤖 Generated by Copilot at 060b900</samp>

Added a feature to preview custom CSS changes for site pages in a new window. Created a `CSSPreviewConnect` component to inject the CSS changes to the site's content and communicate with the preview window. Modified `page.tsx` files to use the component for site index and slug pages. Updated `PostCard.tsx` to preserve URL query parameters for the preview window.

### <samp>🤖 Generated by Copilot at 060b900</samp>

> _Sing, O Muse, of the skillful coder who devised_
> _A wondrous tool for styling sites with custom grace_
> _`CSSPreviewConnect`, the component he contrived_
> _To show the changes live in a separate space_

### WHY

https://discord.com/channels/976854077709369424/1097349373257597040/1134519001645588625

### <samp>🤖 Generated by Copilot at 060b900</samp>

*  Add `CSSPreviewConnect` component to inject CSS changes from preview window to site's content ([link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-81b6f6a35cbf3df1600ee46e2cb30e93f659c85f1cf33846c5f4ffad391323d3R8), [link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-81b6f6a35cbf3df1600ee46e2cb30e93f659c85f1cf33846c5f4ffad391323d3R191), [link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-fb24106e6afcbffa6f2c9fcd050bb5da40db494651f99fa1c97c4f4d2755f3f0R3), [link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-fb24106e6afcbffa6f2c9fcd050bb5da40db494651f99fa1c97c4f4d2755f3f0R34), [link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-f8e4e4b60615c8e509ddaeb8f070297097c37201e96e70d8e99d46dfff9c3c71R1-R47))
*  Add `PreviewButton` component to open preview window with site's content and custom CSS ([link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-ac9b565fa35701100353359989069dacc0a5d669c3729cd138d9423b24a67788L4-R4), [link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-ac9b565fa35701100353359989069dacc0a5d669c3729cd138d9423b24a67788L109-R111), [link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-ac9b565fa35701100353359989069dacc0a5d669c3729cd138d9423b24a67788R120-R206))
*  Add `useGetState` hook to access CSS value without triggering re-render ([link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-ac9b565fa35701100353359989069dacc0a5d669c3729cd138d9423b24a67788R11-R12), [link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-ac9b565fa35701100353359989069dacc0a5d669c3729cd138d9423b24a67788R120-R206))
*  Append current URL search parameters to post link to preserve preview window query parameters ([link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-08902826ffbce5f61f80184493f753baea8584afeccdc639ee3efc86c0ae1dbeL5-R5), [link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-08902826ffbce5f61f80184493f753baea8584afeccdc639ee3efc86c0ae1dbeL43-R51))
*  Remove empty line for code style consistency ([link](https://github.com/Crossbell-Box/xLog/pull/839/files?diff=unified&w=0#diff-ac9b565fa35701100353359989069dacc0a5d669c3729cd138d9423b24a67788L21))

### CLAIM REWARDS

For first-time contributors, please leave your xLog address and Discord ID below to claim your rewards.

### CODE REVIEW

File: src/app/dashboard/[subdomain]/settings/css/page-component.tsx [hunk 1]
```diff
@@ -115,3 +117,90 @@ export default function SettingsCSSPage() {
...
+  const [isInPreview, setIsPreview] = useState(false)
+
+  const handlePreview = () => {
+    let url: URL
+
+    if (IS_DEV) {
+      url = new URL(`http://${subdomain}.localhost:2222/`)
+    } else {
+      url = new URL(location.href)
+      // TODO
```
Comment (Innei) [inline src/app/dashboard/[subdomain]/settings/css/page-component.tsx:140]: I want to know what customize domain or second-level domain name is, and replace this fake URL

File: src/components/site/CSSPreviewConnect.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+"use client"
```
Comment (DIYgod) [inline src/components/site/CSSPreviewConnect.tsx:1]: The saved custom CSS in `src/app/site/[site]/layout.tsx` will still exist and will affect the final result.

File: src/components/site/CSSPreviewConnect.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import Style from "../common/Style"
+
+export const CSSPreviewConnect = () => {
+  const [css, setCss] = useState("")
+
+  useEffect(() => {
+    const search = location.search
+    const searchParams = new URLSearchParams(search)
+
+    let targinOrigin = searchParams.get("origin")
```
Comment (DIYgod) [inline src/components/site/CSSPreviewConnect.tsx:14]: I think we can fix the origin to `SITE_URL` for higher security.

**Review by DIYgod — CHANGES_REQUESTED** [review]

File: src/components/site/CSSPreviewConnect.tsx (hunk 2 above)
Comment (Innei) [reply src/components/site/CSSPreviewConnect.tsx]: Ok, I move to Root Style component to dot this

**Pull request merged**